src/network/networkd-link.c

```diff
@@ -218,6 +233,32 @@ static IPv6PrivacyExtensions link_ipv6_privacy_extensions(Link *link) {
         return link->network->ipv6_privacy_extensions;
 }
 
+static int link_enable_ipv6(Link *link) {
+        const char *p = NULL;
+        bool state = false;
+        int r;
+
+        if (link->flags & IFF_LOOPBACK)
+                return 0;
+
+        if (!link_ipv6_enabled(link))
+                state = true;
+
+        p = strjoina("/proc/sys/net/ipv6/conf/", link->ifname, "/disable_ipv6");
+
+        r = write_string_file(p, one_zero(state), WRITE_STRING_FILE_VERIFY_ON_FAILURE);
+        if (r < 0)
+                log_link_warning_errno(link, r, "Cannot set IPv6 for interface: %m");
+        else {
+                if (state)
+                        log_link_warning_errno(link, r, "IPv6 disabled for interface: %m");
+                else
+                        log_link_warning_errno(link, r, "IPv6 enabled for interface: %m");
+        }
```

Comment:
given that we don't really have an error here it sounds more obvious to use log_link_warning() here, instead of log_link_warning_error(). Of course, this isn't really a bug, since log_link_warning_errno() degrades to log_link_warning() if r is 0 anyway, but it still "feels" like a bug, hence please fix! ;-)